### DISCUSSION TITLE
Adding to state in sglang function

### DISCUSSION THREAD

**Discussion by eaubin** [sgl-project/sglang#209] [General]
Is there a way to add to state that's returned from an sgl function?  I'd like to capture plain python values created during processing and also access variables bound with gen() in forks.

I get `ProgramState' object does not support item assignment` error naively trying:

```
@sgl.function
def test(s):
    s += "Hello"
    s['foo'] = 1
test()
```

**Comment by eaubin** [comment]
FWIW, the method I'm using right now is to store what I need in `s.stream_executor.variables`.

**Comment by hnyls2002** [comment]
@eaubin Yeah, a new PR has been submitted here #198
